Remove redux fetch pt ii

xwatkins opened the pull request.

dlrice commented:
I noticed a few unexpected behaviours:

### 1. Facets persisting when new query is started
Steps to reproduce:
1. Search for `a4` (http://localhost:8080/uniprotkb?query=a4)
2. Click the `reviewed` facet (http://localhost:8080/uniprotkb?query=a4&facets=reviewed:true)
3. The search bar query will still have `a4` but I think that's ok (facets aren't supposed to update the query are they?)
4. Change the search bar query to `mouse` and click search (http://localhost:8080/uniprotkb?facets=reviewed%3Atrue&query=mouse)
5. The facets are still there from the previous search. This is different to the gh-pages behaviour which removes all facets on a new search.


### 2. Loading indicator not immediately shown
Scrolling to the bottom of the first batch of results does not show the loading indicator when loading the second batch. It is visible however if I scroll up a small amount and then back down.

dlrice commented on src/uniprotkb/components/entry/EntryPublicationsFacets.tsx:
```diff
@@ -1,29 +1,40 @@
+    const facet: SelectedFacet = { name, value };
+    history.push({
+      pathname: `/uniprotkb/${accession}/publications`,
+      search: `${facetsAsString([...selectedFacets.concat(facet)])}`,
```
Template string not needed here

dlrice commented on src/uniprotkb/components/entry/EntryPublicationsFacets.tsx:
```diff
@@ -1,29 +1,40 @@
-    );
+    history.push({
+      pathname: `/uniprotkb/${accession}/publications`,
+      search: `${facetsAsString(
```
Template string not needed here

dlrice commented on src/uniprotkb/components/results/utils.ts:
```diff
@@ -0,0 +1,64 @@
+import queryStringModule from 'query-string';
```
UniProtKB utils are now grouped at `./src/uniprotkb/utils/*`

xwatkins replied:
Moved. Although it feels weird to have a generic `utils` with utils files which are then specific to other locations...maybe we can improve this later.

dlrice commented on src/uniprotkb/components/entry/__tests__/__snapshots__/EntryMain.spec.tsx.snap:
```diff
@@ -821,15 +821,6 @@ exports[`Entry view should render 1`] = `
-              <div
-                class="GoRibbon__container__message"
-              >
-                Loading...
```
Just wondering where this went to? I can see the `Loading...` message in the browser.

xwatkins replied:
If this is what I think we don't want to test the GO ribbon component itself so I mocked it (returns an empty object)

dlrice replied:
Ah that's much nicer.

xwatkins replied:
We were getting warnings in the results because of it! 😿

xwatkins replied:
I don't know why I put a crying cat there. Still adjusting to the usage of emojis.

xwatkins commented:
Fixed 1. and addressed other comments.
For 2. I noticed there is a bit of a visual glitch when scrolling and I think it would take time to investigate. As it's minor I'd be more inclined to create a Jira rather than try to address it now.
